plugins/modules/collaborator_information.py

```diff
@@ -0,0 +1,342 @@
+    type: bool
+    returned: only if at least one collaborator is contained within repository
+
+collaborators['<ORG NAME>/<REPO NAME>'].<INDEX>.permissions.triage:
+    description: Will return true if triage rights are given to collaborator.
+                 Triage role can request reviews on pull requests (PRs), mark issues and PRs as duplicates, and add or remove milestones on issues and PRs.
+                 NO WRITE ACCESS.
+    type: bool
+    returned: only if at least one collaborator is contained within repository
+
+collaborators['<ORG NAME>/<REPO NAME>'].<INDEX>.site_admin:
+    description: Will return true if collaborator is a site admin.
+                 This permission gives the collaborator the ability to manage users, organizations, and repositories.
+    type: bool
+    returned: only if at least one collaborator is contained within repository
+
+collaborators['<ORG NAME>/<REPO NAME>'].<INDEX>.type:
+    description: This will return what type of collaborator the user is.
+    type: str
+    returned: only if at least one collaborator is contained within repository
+'''
+
+
+def add_collaborator(g, repo, to_add):
+    r = g.get_repo(repo)
+    for p in to_add:
+        r.add_to_collaborators(p, permission=to_add[p])
+
+
+def check_permissions(g, repo, user_to_check):
+    status = True
+    r = g.get_repo(repo)
+    for user, permission in user_to_check.items():
+        if r.get_collaborator_permission(user) != permission:
+            status = False
+    return status
+
+
+def del_collaborators(g, repo, to_remove):
+    r = g.get_repo(repo)
+    collaborators = r.get_collaborators(affiliation="direct")
+    for collaborator in collaborators:
+        if collaborator.login in to_remove:
+            r.remove_from_collaborators(collaborator.login)
+
+
+def change_collaborator_permissions(g, repo, to_change):
+    r = g.get_repo(repo)
+    collaborator_set = set()
+    collaborators = r.get_collaborators(affiliation="direct")
+    for collaborator in collaborators:
+        collaborator_set.add(collaborator.login)
+    for p in to_change:
+        if p in collaborator_set:
+            r.add_to_collaborators(p, permission=to_change[p])
+
+
+def get_collaborators(g, repo):
+    output = list()
+    collab_output = dict()
+    collaborators = g.get_repo(repo).get_collaborators(affiliation="direct")
+    for collaborator in collaborators:
+        collab_output['login'] = collaborator.login
+        collab_output['id'] = collaborator.id
+        collab_output['type'] = collaborator.type
+        collab_output['site_admin'] = collaborator.site_admin
+        permissions = {
+            'triage': collaborator.permissions.triage,
+            'push': collaborator.permissions.push,
+            'pull': collaborator.permissions.pull,
+            'admin': collaborator.permissions.admin
+        }
+        collab_output['permissions'] = permissions
+
+        output.append(collab_output)
+
+    return output
+
```

Comment:
This code does not work.  Please fix the bug and add unit tests that would have found the bug.

Reply:
output will be populated with objects that have the same values for all fields.